### PR TITLE
Add init for direct database property

### PR DESCRIPTION
**Pull request by Jasperav**
Fixes: https://github.com/groue/GRDBQuery/issues/6

### CODE REVIEW

File: Documentation/QueryDemo/QueryDemo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -78,7 +78,7 @@
 			isa = PBXFrameworksBuildPhase;
 			buildActionMask = 2147483647;
 			files = (
-				56519DC0274FC3F700ED16D8 /* GRDB in Frameworks */,
+				69BAF48A27AAB84800452053 /* GRDB in Frameworks */,
```
Comment (Jasperav) [inline Documentation/QueryDemo/QueryDemo.xcodeproj/project.pbxproj:81]: This are some weird things Xcode has done, it was stuck on package resolving... now it uses the latest GRDB version, I can revert this changes if you want though

**Comment by groue** [comment]
Hello @Jasperav. This pull request is very generous, but I do not feel comfortable with it.

Yes, I could try it with Xcode 13.2.1 and the demo app works.

Yet [`@Environment`](https://developer.apple.com/documentation/swiftui/environment) is clearly designed to be used as a property wrapper, and doing otherwise is just taking a risk. I'm much too stupid for playing a "who's the smartest" game with SwiftUI.

In other words, _I do not trust_ Apple for 1. testing the behavior of their public SwiftUI apis when used outside of the tracks, and 2. preserving the undocumented invariants some apps are fool enough to rely on. The fact that your changes work today is no guarantee that they will work in the future. And who will pay? Me, because I'll have GitHub issues to answer. And most importantly, the poor users who will have to perform a painful refactoring, or disfigure their app with gross workarounds.

You know what I mean: painful refactoring, and application disfiguration ARE the price to pay when one bets against SwiftUI, and loses. Experience leaves no doubt about it.

[You wrote](https://github.com/groue/GRDBQuery/issues/6):

> My views already hold an object in which the database context lives, so I was wondering if I can initialize a property annotated with `@Query` in the `init` block.

Did you consider having those objects grab the database from the environment instead, now that you found a technique to do so (from `Environment(keyPath).wrappedValue`)?

**Comment by groue** [comment]
I totally agree with the answers you got in the Swift forums. Setting the underscored property from `init()` is 100% supported by the language.

But this is not what this PR does, and the question you asked in the forums is unrelated.

This PR does this:

```swift
public struct Query<Request: Queryable>: DynamicProperty {
    private var database: Request.DatabaseContext
    public init(
         _ request: Request,
         in keyPath: KeyPath<EnvironmentValues, Request.DatabaseContext>)
     {
         // Here be dragons
         database = Environment(keyPath).wrappedValue
         initialRequest = request
     }
}
```

This `Environment(keyPath).wrappedValue` expression gives me the chills. It relies on the SwiftUI magic in a way that makes me deeply uncomfortable.

What does the sample code below prints?

```swift
DispatchQueue.main.async {
    let value = Environment(keyPath).wrappedValue
    print(value)
}
```

Which database is the following query using?

```swift
Button("I'm lucky") {
    let query = Query(...)
}
```

You can make guesses. You can even make educated guesses by running this code, and make conclusions based on what you see. But those are only guesses, and nothing in the documentation tells you that the behavior you see can be relied on.

The only Swift syntax I trust, given the insane amount of magic SwiftUI relies on, is `@Environment var database` and `@Query var myValue` declarations in types that conform to blessed SwiftUI protocols. Everything else is playing russian roulette.

Maybe my experience will grow, and I'll start to rely on some SwiftUI internals. This time has not come yet: I'm still a young and very conservative SwiftUI developer (at least in the code I release publicly).

Nothing prevents _you_ from performing experiments, though!

**Comment by groue** [comment]
> > Did you consider having those objects grab the database from the environment instead, now that you found a technique to do so
> 
> What do you mean with this?

Since you can feed a `Query` from the environment with `Environment(\.database).wrappedValue` (as you did in this PR), you can feed a view model from `Environment(\.database).wrappedValue` as well, instead of feeding `Query` from the view model's database.

**Comment by Jasperav** [comment]
@groue Alright, fair play. https://developer.apple.com/documentation/swiftui/environment/wrappedvalue indeeds mention that the `wrappedValue` shouldn't be accessed directly. Maybe it will change in the future. 

In the mean time I will use my fork :)

**Pull request closed without merging**